wagtail/documents/views/chooser.py

```diff
@@ -51,157 +67,186 @@ def get_cell_context_data(self, instance, parent_context):
-                "title",
-                label=_("Title"),
-                url_name="wagtaildocs:document_chosen",
-                link_attrs={"data-chooser-modal-choice": True},
-            ),
+        if len(collections) < 2:
+            return None
+
+        return collections
+
+    @property
+    def columns(self):
+        columns = super().columns + [
             DownloadColumn("filename", label=_("File")),
             DateColumn("created_at", label=_("Created"), width="16%"),
         ]
 
         if self.collections:
             columns.insert(2, Column("collection", label=_("Collection")))
 
-        self.table = Table(columns, self.documents)
-
-        return self.render_to_response()
-
-    def get_context_data(self):
-        return {
-            "documents": self.documents,
-            "documents_exist": self.documents_exist,
-            "table": self.table,
-            "uploadform": self.uploadform,
-            "query_string": self.q,
-            "searchform": self.searchform,
-            "collections": self.collections,
-            "is_searching": self.is_searching,
-            "collection_id": self.collection_id,
-        }
-
-    def render_to_response(self):
-        raise NotImplementedError()
-
-
-class ChooseView(BaseChooseView):
-    def render_to_response(self):
-        return render_modal_workflow(
-            self.request,
-            "wagtaildocs/chooser/chooser.html",
-            None,
-            self.get_context_data(),
-            json_data={
-                "step": "choose",
-                "tag_autocomplete_url": reverse("wagtailadmin_tag_autocomplete"),
-            },
-        )
+        return columns
 
+    def get(self, request):
+        self.model = get_document_model()
+        self.collection_id = None
 
-class ChooseResultsView(BaseChooseView):
-    def render_to_response(self):
-        return TemplateResponse(
-            self.request, "wagtaildocs/chooser/results.html", self.get_context_data()
-        )
+        return super().get(request)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["collection_id"] = self.collection_id
+        return context
+
+
+class DocumentChooseViewMixin(ChooseViewMixin):
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["collections"] = self.collections
+        return context
+
+    def get_response_json_data(self):
+        json_data = super().get_response_json_data()
+        json_data["tag_autocomplete_url"] = reverse("wagtailadmin_tag_autocomplete")
+        return json_data
+
+
+class ChooseView(
+    DocumentChooseViewMixin, DocumentCreationFormMixin, BaseDocumentChooseView
+):
+    pass
+
+
+class ChooseResultsView(
+    ChooseResultsViewMixin, DocumentCreationFormMixin, BaseDocumentChooseView
+):
+    pass
+
+
+class DocumentChosenView(ChosenViewMixin, DocumentChosenResponseMixin, View):
```

Comment:
question: the other views don't have the `Document` prefix. Worth either dropping it here, or adding it to the other ones (`ChooseView`, `ChooseResultsView`) for consistency?

Reply:
I know, I know. Hence more of a question than a "thou shalt do so" statement :)
My thinking was along the lines of any import (should someone brave enough to go and customize everything). Going back to hiding (-: